util/chunk/column.go

```diff
@@ -246,6 +246,64 @@ const (
 	sizeMyDecimal = int(unsafe.Sizeof(types.MyDecimal{}))
 )
 
+// preAlloc allocates space for a fixed-length-type slice and resets all slots to null.
+func (c *Column) preAlloc(length, typeSize int) {
+	c.nullCount = 0
+	nData := length * typeSize
+	if len(c.data) > nData {
+		c.data = c.data[:nData]
+	} else {
+		c.data = make([]byte, nData)
+	}
+
+	nBitmap := (length + 7) >> 3
+	if len(c.nullBitmap) > nBitmap {
+		c.nullBitmap = c.nullBitmap[:nBitmap]
+		for i := range c.nullBitmap {
+			// resets all slots to null.
+			c.nullBitmap[i] = 255
+		}
+	} else {
+		c.nullBitmap = make([]byte, nBitmap)
+	}
+
+	c.length = length
```

Comment:
Should we reset the other attributes for `c`?

Reply:
For a fixed-length type, the field `offset` is not used and `elemBuf` doesn't matter so we ignore them while all other fields are reset in this method.

Reply:
Check https://github.com/pingcap/tidb/blob/master/util/chunk/chunk.go#L341

Reply:
My fault, you are right.
I will fix it soon.

Reply:
fixed.